static/js/compose.js

```diff
@@ -441,6 +441,9 @@ export function initialize() {
     $("#compose-textarea").on("input propertychange", () => {
         compose_validate.warn_if_topic_resolved(false);
         const compose_text_length = compose_validate.check_overflow_text();
+        if (compose_text_length !== 0 && $("#compose-textarea").hasClass("invalid")) {
```

Comment:
This check is in a hot path -- we call this function on every character press. It's probably not ideal to be doing a DOM query rather than just using a JavaScript variable to keep track of this state, but I couldn't see this jQuery item in a profile (probably because it is an ID lookup), so I'm OK merging this.

Reply:
it should exit early if the box isn't empty, right?

Reply:
Yes but if you're typing a lot of characters into the box, on most keypresses, the box will not be empty. The point is I think we run this on every key you press while typing, and one needs to be very careful with any code that does that, since anything that's unexpectedly slow in such a code path (for example, something that causing a paint to happen) can make typing freeze or feel laggy.

Reply:
> it should exit early if the box isn't empty, right?

oh I'm silly, it exists early if the box _is_ empty, which doesn't help. Yeah that makes a lot of sense, thanks for calling it out!